src/main/java/gregtech/api/capability/impl/miner/MultiblockMinerLogic.java

```diff
@@ -0,0 +1,148 @@
+        return super.checkShouldStop() || !miner.drainFluid(true);
+    }
+
+    @Override
+    protected void drainStorages() {
+        super.drainStorages();
+        miner.drainFluid(false);
+    }
+
+    @Override
+    protected void getSmallOreBlockDrops(NonNullList<ItemStack> blockDrops, WorldServer world, BlockPos blockToMine, IBlockState blockState) {
+        // Small ores: use (fortune bonus + overclockAmount) value here for fortune, since every overclock increases the yield for small ores
+        super.getSmallOreBlockDrops(blockDrops, world, blockToMine, blockState);
+    }
+
+    @Override
+    protected void getRegularBlockDrops(NonNullList<ItemStack> blockDrops, WorldServer world, BlockPos blockToMine, IBlockState blockState) {
+        if (!isSilkTouchMode) // 3X the ore compared to the single blocks
+            applyTieredHammerNoRandomDrops(world.rand, blockState, blockDrops, 3, null, this.blockDropRecipeMap, this.voltageTier);
+        else
+            super.getRegularBlockDrops(blockDrops, world, blockToMine, blockState);
+    }
+
+    @Override
+    public void initPos(BlockPos pos, int currentRadius) {
+        if (!isChunkMode) {
+            super.initPos(pos, currentRadius);
+        } else {
+            WorldServer world = (WorldServer) this.metaTileEntity.getWorld();
+            Chunk origin = world.getChunk(this.metaTileEntity.getPos());
+            ChunkPos startPos = (world.getChunk(origin.x - currentRadius / CHUNK_LENGTH, origin.z - currentRadius / CHUNK_LENGTH)).getPos();
+            getX().set(startPos.getXStart());
+            getY().set(this.metaTileEntity.getPos().getY() - 1);
+            getZ().set(startPos.getZStart());
+            getStartX().set(startPos.getXStart());
+            getStartZ().set(startPos.getZStart());
+            getMineX().set(startPos.getXStart());
+            getMineY().set(this.metaTileEntity.getPos().getY() - 1);
+            getMineZ().set(startPos.getZStart());
+            getStartY().set(this.metaTileEntity.getPos().getY());
+            getPipeY().set(this.metaTileEntity.getPos().getY() - 1);
+        }
+    }
+
+    public void setVoltageTier(int tier) {
+        this.voltageTier = tier;
+    }
+
+    public void setOverclockAmount(int amount) {
+        this.overclockAmount = amount;
+    }
+
+    public int getOverclockAmount() {
+        return this.overclockAmount;
+    }
+
+    public boolean isChunkMode() {
+        return this.isChunkMode;
+    }
+
+    public void setChunkMode(boolean isChunkMode) {
+        if (isActive()) {
+            getX().set(Integer.MAX_VALUE);
+            getY().set(Integer.MAX_VALUE);
+            getZ().set(Integer.MAX_VALUE);
+            this.isChunkMode = isChunkMode;
+        }
+    }
+
+    public boolean isSilkTouchMode() {
+        return this.isSilkTouchMode;
+    }
+
+    public void setSilkTouchMode(boolean isSilkTouchMode) {
+        this.isSilkTouchMode = isSilkTouchMode;
+    }
```

Comment:
This should have a check for if the machine is not active, to match the tooltip of the button.